.github/workflows/ci-master.yml

```diff
@@ -50,29 +46,28 @@ jobs:
       matrix:
         node-version: [18.x, 20.x]
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v3.5.3
         with:
           repository: n8n-io/n8n
           ref: ${{ inputs.branch }}
 
-      - name: Restore cached build artifacts
-        uses: actions/cache@v3
-        with:
-          path: |
-            /github/home/.cache
-            /github/home/.pnpm-store
-            ./node_modules
-            ./packages
-          key: ${{ github.sha }}-base:${{ matrix.node-version }}-test-lint
-
-      - uses: pnpm/action-setup@v2.2.4
+      - uses: pnpm/action-setup@v2.4.0
 
       - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v3
+        uses: actions/setup-node@v3.7.0
         with:
           node-version: ${{ matrix.node-version }}
           cache: pnpm
 
+      - name: Install dependencies
+        run: pnpm install --frozen-lockfile
```

Comment:
Since this is hardlinking to the store, I take it this doesn't add much overhead vs. caching as we did before.

Reply:
yeah. this also has the advantage that if there are random 500x from the caching service, CI still works, making the setup a bit more resilient.